parser/prism/Translator.cc

```diff
@@ -306,7 +306,14 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
 
             auto expr = translate(blockArg->expression);
 
-            return make_unique<parser::BlockPass>(location, move(expr));
+            if (!directlyDesugar || !hasExpr(expr)) {
+                return make_unique<parser::BlockPass>(location, std::move(expr));
+            }
+
+            // TODO: we copy here to prevent segfaults, but why are we getting segfaults with takeDesugaredExpr?
+            auto desugaredExpr = expr == nullptr ? MK::EmptyTree() : expr->copyDesugaredExpr();
+            auto blockArgNode = MK::BlockArg(location, move(desugaredExpr));
```

Comment:
Arrrghhh, bad naming strikes again. `BlockArg` represents a block _parameter_, as in `def foo(&this_one)`, not `method_call(&block_pass)`.

There is no direct `ast::ExpressionPtr` representation of a `BlockPass`. It's handled specially in the `parser::Send *` case of the Desugarer. We'll need intrusive changes in `DesugarPrism.cc` for this.